packages/astro/src/core/config/schema.ts

```diff
@@ -140,7 +140,6 @@ export const AstroConfigSchema = z.object({
 				.optional()
 				.default(ASTRO_CONFIG_DEFAULTS.build.excludeMiddleware),
 		})
-		.optional()
```

Comment:
Don't we need this?

Reply:
After doing some research, I guess not! `.optional()` + `.default({})` is redundant. There will always be a value.

Reply:
Yup, this was a surprise for me as well! I wouldn't have made a change in this PR except that it actually changes the schema object's API when I wanted to get the shape of the schema back out for validation.